app/controllers/mixins/ems_common_angular.rb

```diff
@@ -420,6 +423,12 @@ def set_ems_record_vars(ems, mode = nil)
       build_connection(ems, endpoints, mode)
     end
 
+    def get_hostname_from_routes(ems, hostname, port, token)
+      client = ems.class.raw_connect(hostname, port, :bearer => token)
```

Comment:
> why going to hawkular? We should have this information in our db.

@simon3z this area of the code is called also after pressing the authenticate button in the ui, we do not yet have this informations, we get it just after refresh.

IMHO it's ok to ask kubclient for the routes because we are doing authentication also by communicating with kubeclient.